src/NServiceBus.Hosting.Windows/StartableAndStoppable/StartableAndStoppableRunner.cs

```diff
@@ -96,8 +109,29 @@ public Task Start(IMessageSession session)
             }
         }
 
+        CancellationTokenSource LongRunningWarning(string message)
+        {
+            var tokenSource = new CancellationTokenSource();
+
+            // ReSharper disable once MethodSupportsCancellation
+            Task.Run(async () =>
```

Comment:
You don't need to wrap this in Task.Run. Add a continuation to Task.Delay which contains the Warn and only gets called when the antecedent task successfully completed.